cmd/minishift/cmd/root.go

```diff
@@ -46,7 +48,7 @@ import (
 )
 
 var dirs = [...]string{
```

Comment:
I don't think we can keep the initialisation like this. The vars are getting initialised before we determine the profile to use. This is probably also the reason why you later have to reset the directories. Really, we need to do the following 

* Initialise Viper
* Determine the profile
* Initialise the required profile specific data structures/directories

I recommend to put the values of _dirs_ into a struct. This way the required directories can be referred to in a type-safe manner. Using _dirs[0] = "bla"_, etc is very flaky.